test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -312,6 +312,17 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
         val (z, caseValue) = f(z0, t)
         ZIO.succeed(z -> Spec(caseValue))
     }
+
+  /**
+   * Runs only tests whose labels (which must be strings) contain the given substring
```

Comment:
We should probably clarify the contract with regard to suites. "Returns a new spec containing only specs with labels containing the specified string. If a suite label contains the specified string all specs in that suite will be included in the resulting spec. This method is only available on specs whose label type is `String`.